### ISSUE TITLE
EKF2 not merging the Rangefinder

### ISSUE THREAD

**Issue by JMshare** [PX4/Firmware#12247]
**Describe the bug**
The estimated altitude and the local_position.z seem to ignore the rangefinder data.
Using the Lidar Lite through PWM,
Pixhawk 4 fmu_v5,
stable release 1.9.0,
in position control or altitude hold mode.

The distance sensor data show that the sensor outputs correct readings but the altitude estimates seem uninfluenced.

**Log Files and Screenshots**
https://review.px4.io/plot_app?log=d19babaf-06b4-4074-b14e-af8828fb51c2

Add screenshots to help explain your problem.
![Screenshot 2019-06-11 at 20 51 47](https://user-images.githubusercontent.com/22155994/59302939-072a4980-8c8d-11e9-9eeb-36fe5cfd2355.png)

**Comment by gajena** [comment]
@JMshare Did you find any solution for this? I am also having the same problem with fmu_v3 and LidarLite.

**Comment by JMshare** [comment]
No, I don't really know how to fix this, hoped someone from the developers would react.

But if it helps to track the issue, the reported _**signal_quality 0**_ in the log must be the reason why the ekf ignores the data. Also I remember somehow coming to a conclusion that this is related to LidarLite specifically, not just any rangefinder. The newest model had some changes in the firmware that haven't been accounted for in the PX4 yet or something like that...

**Comment by gajena** [comment]
I have tested the same LiDaR with v1.8.1, and it was working fine. So after digging in the firmware, I found the heck.https://github.com/AerialRobotics-IITK/Firmware/commit/25e0d01e3a869651d2cb21ba1477ae0c30fee030

**Comment by JMshare** [comment]
Thanks @gajena, I was thinking about doing the same thing until the driver is fixed.

**Comment by tecdee** [comment]
@gajena sorry if the question is silly but i'm quite new to px4 and all, so how do I make use of the hack?
- Modify the 'ekf2_main.cpp' file as per the hack from the link you gave
- build the firmware, and load as custom firmware.
Am I right ?
Thanks for your help.

**Comment by gajena** [comment]
Yes or just pull from my repo(please pull latest commit).

**Comment by tecdee** [comment]
> Yes or just pull from my repo(please pull latest commit).

Thank you @gajena
